cvat/apps/engine/background_operations.py

```diff
@@ -0,0 +1,666 @@
+# Copyright (C) 2024 CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import os
+import os.path as osp
+import cvat.apps.dataset_manager as dm
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, Callable, Dict, Optional, Union
+
+import django_rq
+from attrs.converters import to_bool
+from django.conf import settings
+from django.http.request import HttpRequest
+from django.utils import timezone
+from django_rq.queues import DjangoRQ
+from rest_framework import serializers, status
+from rest_framework.response import Response
+from rest_framework.reverse import reverse
+from rq.job import Job as RQJob
+from rq.job import JobStatus as RQJobStatus
+
+from cvat.apps.engine import models
+from cvat.apps.engine.backup import ProjectExporter, TaskExporter, create_backup
+from cvat.apps.engine.cloud_provider import export_resource_to_cloud_storage
+from cvat.apps.engine.location import StorageType, get_location_configuration
+from cvat.apps.engine.log import ServerLogManager
+from cvat.apps.engine.models import Location, Project, Task
+from cvat.apps.engine.permissions import get_cloud_storage_for_import_or_export
+from cvat.apps.engine.rq_job_handler import RQIdManager
+from cvat.apps.engine.serializers import RqIdSerializer
+from cvat.apps.engine.utils import (
+    build_annotations_file_name,
+    build_backup_file_name,
+    define_dependent_job,
+    get_rq_job_meta,
+    get_rq_lock_by_user,
+    sendfile,
+)
+from cvat.apps.events.handlers import handle_dataset_export
+
+slogger = ServerLogManager(__name__)
+
+
+class _ResourceExportManager(ABC):
+    QUEUE_NAME = settings.CVAT_QUEUES.EXPORT_DATA.value
+
+    def __init__(
+        self,
+        version: int,
+        db_instance: Union[models.Project, models.Task, models.Job],
+        *,
+        callback: Callable,
+    ) -> None:
+        self.version = version
+        self.db_instance = db_instance
+        self.obj_type = db_instance.__class__.__name__.lower()
+        if self.obj_type not in self.SUPPORTED_RESOURCES:
+            raise ValueError(
+                "Unexpected type of db_instance: {}".format(type(db_instance))
+            )
+
+        self.callback = callback
```

Comment:
I suggest that this field is either renamed or any documentation is added to clarify it's purpose. It's not used in the base class and the name itself is quite generic.